# Machine Learning Lab-2 (100 points)

## 1) Load the Diabetes dataset into a dataframe (50 points)

You can find the diabetes dataset at https://scikit-learn.org/stable/modules/generated/sklearn.datasets.load_diabetes.html#sklearn.datasets.load_diabetes.
You can use the Boston dataset file as a reference

### 1.1) Load the dataset (10 points)

In [95]:
from sklearn.datasets import load_diabetes

diabetes_data=load_diabetes()


### 1.2) Print a description of the data (10 points)

In [96]:

print(diabetes_data.DESCR)



.. _diabetes_dataset:

Diabetes dataset
----------------

Ten baseline variables, age, sex, body mass index, average blood
pressure, and six blood serum measurements were obtained for each of n =
442 diabetes patients, as well as the response of interest, a
quantitative measure of disease progression one year after baseline.

**Data Set Characteristics:**

  :Number of Instances: 442

  :Number of Attributes: First 10 columns are numeric predictive values

  :Target: Column 11 is a quantitative measure of disease progression one year after baseline

  :Attribute Information:
      - age     age in years
      - sex
      - bmi     body mass index
      - bp      average blood pressure
      - s1      tc, T-Cells (a type of white blood cells)
      - s2      ldl, low-density lipoproteins
      - s3      hdl, high-density lipoproteins
      - s4      tch, thyroid stimulating hormone
      - s5      ltg, lamotrigine
      - s6      glu, blood sugar level

Note: Each of these 10 feature va

### 1.3) Write a small description of the Diabetes dataset. You can use the output of the previous cell for your reference (10 points)

.. _diabetes_dataset:

Diabetes dataset
----------------

Ten baseline variables, age, sex, body mass index, average blood
pressure, and six blood serum measurements were obtained for each of n =
442 diabetes patients, as well as the response of interest, a
quantitative measure of disease progression one year after baseline.

**Data Set Characteristics:**

  :Number of Instances: 442

  :Number of Attributes: First 10 columns are numeric predictive values

  :Target: Column 11 is a quantitative measure of disease progression one year after baseline

  :Attribute Information:
      - age     age in years
      - sex
      - bmi     body mass index
      - bp      average blood pressure
      - s1      tc, T-Cells (a type of white blood cells)
      - s2      ldl, low-density lipoproteins
      - s3      hdl, high-density lipoproteins
      - s4      tch, thyroid stimulating hormone
      - s5      ltg, lamotrigine
      - s6      glu, blood sugar level

Note: Each of these 10 feature variables have been mean centered and scaled by the standard deviation times `n_samples` (i.e. the sum of squares of each column totals 1).

Source URL:
https://www4.stat.ncsu.edu/~boos/var.select/diabetes.html

For more information see:
Bradley Efron, Trevor Hastie, Iain Johnstone and Robert Tibshirani (2004) "Least Angle Regression," Annals of Statistics (with discussion), 407-499.
(https://web.stanford.edu/~hastie/Papers/LARS/LeastAngle_2002.pdf)


### 1.4) Create a dataframe and import the diabetes data (input) to the dataframe. Print the dataframe that you have created (10 points)

In [116]:
import pandas as pan

dataFr=pan.DataFrame(data=diabetes_data.data, columns=diabetes_data.feature_names)

print(dataFr)




          age       sex       bmi        bp        s1        s2        s3  \
0    0.038076  0.050680  0.061696  0.021872 -0.044223 -0.034821 -0.043401   
1   -0.001882 -0.044642 -0.051474 -0.026328 -0.008449 -0.019163  0.074412   
2    0.085299  0.050680  0.044451 -0.005671 -0.045599 -0.034194 -0.032356   
3   -0.089063 -0.044642 -0.011595 -0.036656  0.012191  0.024991 -0.036038   
4    0.005383 -0.044642 -0.036385  0.021872  0.003935  0.015596  0.008142   
..        ...       ...       ...       ...       ...       ...       ...   
437  0.041708  0.050680  0.019662  0.059744 -0.005697 -0.002566 -0.028674   
438 -0.005515  0.050680 -0.015906 -0.067642  0.049341  0.079165 -0.028674   
439  0.041708  0.050680 -0.015906  0.017282 -0.037344 -0.013840 -0.024993   
440 -0.045472 -0.044642  0.039062  0.001215  0.016318  0.015283 -0.028674   
441 -0.045472 -0.044642 -0.073030 -0.081414  0.083740  0.027809  0.173816   

           s4        s5        s6  
0   -0.002592  0.019908 -0.017646  
1  

### 1.5) Print the target of the diabetes data (10 points)

In [117]:

print(diabetes_data.target)



[151.  75. 141. 206. 135.  97. 138.  63. 110. 310. 101.  69. 179. 185.
 118. 171. 166. 144.  97. 168.  68.  49.  68. 245. 184. 202. 137.  85.
 131. 283. 129.  59. 341.  87.  65. 102. 265. 276. 252.  90. 100.  55.
  61.  92. 259.  53. 190. 142.  75. 142. 155. 225.  59. 104. 182. 128.
  52.  37. 170. 170.  61. 144.  52. 128.  71. 163. 150.  97. 160. 178.
  48. 270. 202. 111.  85.  42. 170. 200. 252. 113. 143.  51.  52. 210.
  65. 141.  55. 134.  42. 111.  98. 164.  48.  96.  90. 162. 150. 279.
  92.  83. 128. 102. 302. 198.  95.  53. 134. 144. 232.  81. 104.  59.
 246. 297. 258. 229. 275. 281. 179. 200. 200. 173. 180.  84. 121. 161.
  99. 109. 115. 268. 274. 158. 107.  83. 103. 272.  85. 280. 336. 281.
 118. 317. 235.  60. 174. 259. 178. 128.  96. 126. 288.  88. 292.  71.
 197. 186.  25.  84.  96. 195.  53. 217. 172. 131. 214.  59.  70. 220.
 268. 152.  47.  74. 295. 101. 151. 127. 237. 225.  81. 151. 107.  64.
 138. 185. 265. 101. 137. 143. 141.  79. 292. 178.  91. 116.  86. 122.
  72. 

## 2) Split into Training and Testing data and print the training and testing data (50 points)

### 2.1) Split into Training and Testing data with training to testing ratio 80:20 (10 points)

In [118]:
from sklearn.model_selection import train_test_split

x_train,x_test,y_train,y_test = train_test_split(diabetes_data.data, diabetes_data.target, train_size=0.8)



### 2.2) print x_train (10 points)

In [121]:
print(x_train)


[[ 0.01991321 -0.04464164  0.00457217 ... -0.03949338 -0.02364456
  -0.04664087]
 [ 0.04170844  0.05068012  0.07139652 ...  0.03430886  0.07341008
   0.08590655]
 [ 0.01628068  0.05068012 -0.046085   ... -0.00259226 -0.0439854
  -0.04249877]
 ...
 [-0.02730979 -0.04464164 -0.06332999 ... -0.0763945  -0.05615757
  -0.06735141]
 [ 0.05260606  0.05068012 -0.02452876 ... -0.00259226  0.02671426
  -0.03835666]
 [ 0.01628068 -0.04464164 -0.04500719 ... -0.0763945  -0.0425721
   0.04034337]]


### 2.3) Print y_train (10 points)

In [122]:


print(y_train)



[ 59. 295.  47. 146. 104. 253. 219. 346.  53. 161.  83. 276.  97.  51.
 145. 341.  91.  25.  49. 246. 162. 178.  64.  81. 237.  63. 265. 248.
 141. 221. 150. 170.  77. 143.  91. 179. 242. 164.  31.  85.  65.  59.
 248. 200. 172. 121. 288.  85. 107.  42. 210. 297. 108.  78. 178.  97.
 101. 233.  65. 140.  59. 115. 225. 192.  72. 164. 185. 209. 232.  42.
  99. 129. 268. 214.  78. 182. 134. 197. 178. 214. 243.  87. 200.  86.
  66. 190.  63. 163. 113. 293.  69.  67.  64.  39. 128. 168. 258. 134.
 321. 131. 122. 292. 280. 237. 277. 200. 131. 154. 281. 235. 131.  60.
 306. 138. 156. 144.  53. 233. 135.  90.  68. 102. 142. 216. 265.  75.
 177. 270. 215.  55. 129. 208.  83. 200.  84. 155. 281. 189. 155.  48.
 109. 242. 168.  42. 275. 259. 265. 268. 222.  63.  53. 101.  49. 135.
 200.  71. 235.  66.  52. 202. 201. 104. 118. 252.  55. 179. 140. 170.
 123. 196. 202. 317. 257. 264. 214.  45. 195. 198. 310. 198. 152. 153.
  91. 151. 217. 185. 142.  72. 336. 186. 150.  97. 128. 232.  96.  72.
  77. 

### 2.4) Print x_test (10 points)

In [123]:
print(x_test)



[[ 0.06350368  0.05068012 -0.00189471  0.06662967  0.09061988  0.10891438
   0.02286863  0.01770335 -0.03581673  0.00306441]
 [-0.02730979 -0.04464164 -0.01806189 -0.04009932 -0.00294491 -0.01133463
   0.03759519 -0.03949338 -0.00894402 -0.05492509]
 [ 0.06350368 -0.04464164  0.01750591  0.02187235  0.00806271  0.02154596
  -0.03603757  0.03430886  0.01990842  0.01134862]
 [-0.03457486  0.05068012 -0.05578531 -0.01599922 -0.00982468 -0.00789
   0.03759519 -0.03949338 -0.05295879  0.02791705]
 [-0.06000263  0.05068012  0.05415152 -0.01944209 -0.04972731 -0.04891244
   0.02286863 -0.03949338 -0.0439854  -0.0052198 ]
 [ 0.00538306  0.05068012  0.03043966  0.08384403 -0.03734373 -0.0473467
   0.01550536 -0.03949338  0.00864028  0.01549073]
 [ 0.0562386  -0.04464164 -0.05794093 -0.00796586  0.0520932   0.04910302
   0.05600338 -0.02141183 -0.02832024  0.04448548]
 [ 0.0090156  -0.04464164  0.05522933 -0.00567061  0.05759701  0.04471895
  -0.00290283  0.02323852  0.05568355  0.10661708]
 [ 0

### 2.5) Print y_test (10 points)

In [125]:


print(y_test)



[ 63. 168. 173.  39. 187. 202. 158. 173.  65. 262.  71. 151. 101. 110.
  55. 259. 109. 163. 310. 142. 144. 166. 100. 230. 139. 263. 245.  87.
  61.  70.  77.  69. 277. 103. 311.  84.  93.  74.  94. 138. 252. 220.
 127.  52. 120. 283.  85. 183. 137. 198. 174.  77.  44. 167. 182. 178.
  55. 200. 248. 270.  78. 242.  48. 170.  71. 150. 147.  72.  88. 160.
  83. 144. 302. 122. 111. 274. 281.  48. 178. 217.  60.  94. 310. 136.
 190. 152.  90. 151.  73.]
